components/org.wso2.carbon.identity.authz.spicedb/src/main/java/org/wso2/carbon/identity/authz/spicedb/handler/spicedb/SpicedbPermissionRequestService.java

```diff
@@ -0,0 +1,145 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.authz.spicedb.handler.spicedb;
+
+import com.google.gson.Gson;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.wso2.carbon.identity.authorization.framework.model.AccessEvaluationRequest;
+import org.wso2.carbon.identity.authorization.framework.model.AccessEvaluationResponse;
+import org.wso2.carbon.identity.authorization.framework.model.BulkAccessEvaluationRequest;
+import org.wso2.carbon.identity.authorization.framework.model.BulkAccessEvaluationResponse;
+import org.wso2.carbon.identity.authorization.framework.service.AccessEvaluationService;
+import org.wso2.carbon.identity.authz.spicedb.constants.SpiceDbApiConstants;
+import org.wso2.carbon.identity.authz.spicedb.handler.exception.SpicedbEvaluationException;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.BulkCheckPermissionRequest;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.BulkCheckPermissionResponse;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.CheckPermissionRequest;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.CheckPermissionResponse;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.SpiceDbErrorResponse;
+import org.wso2.carbon.identity.authz.spicedb.handler.util.HttpHandler;
+
+import java.io.IOException;
+
+/**
+ * The {@code SpicedbPermissionRequestService} handles sending all evaluation related requests to SpiceDB
+ * authorization engine.
+ * <p>
+ *     This class implements the {@link AccessEvaluationService} and provides the implementation for access evaluation
+ *     using SpiceDB. This class uses the {@link HttpHandler} to send HTTP requests to the SpiceDB authorization
+ *     engine and parse the responses to generic response models.
+ * </p>
+ */
+public class SpicedbPermissionRequestService implements AccessEvaluationService {
+
+    private final Gson gson = new Gson();
+
+    /**
+     * This method returns the name of the authorization engine.
+     *
+     * @return The name of the authorization engine.
+     */
+    @Override
+    public String getEngine() {
+
+        return SpiceDbApiConstants.AUTHORIZATION_ENGINE_NAME;
+    }
+
+    /**
+     * This method sends a permission check request to SpiceDB and returns the response as an
+     * {@link AccessEvaluationResponse}.
+     *
+     * @param accessEvaluationRequest The request object containing the permission check details.
+     * @return The response object containing the authorization check result.
+     * @throws SpicedbEvaluationException If an error occurs while sending the request or parsing the response.
+     */
+    @Override
+    public AccessEvaluationResponse evaluate(AccessEvaluationRequest accessEvaluationRequest)
+            throws SpicedbEvaluationException {
+
+        CheckPermissionRequest checkPermissionRequest = new CheckPermissionRequest(accessEvaluationRequest);
```

Comment:
Its better if we have some check before start the actual logic, see if there is any that needs to be here